File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter04/IIS Host/MathLibrary/SimpleMath.cs
using System;
using System.Threading;
using System.Diagnostics;

namespace MathLibrary
{  
   public class SimpleMath : MarshalByRefObject
   {
      public SimpleMath()
      {
         WriteLogEntry("SimpleMath ctor called");
      }

      public int Add(int n1, int n2)
      {
         WriteLogEntry(string.Format("SimpleMath.Add({0},{1})", n1, n2));
         return n1 + n2;
      }

      public int Subtract(int n1, int n2)
      {
         WriteLogEntry(string.Format("SimpleMath.Subtract({0},{1})", n1, n2));
         return n1 - n2;
      }

      private void WriteLogEntry(string msg)
      {
         EventLog.WriteEntry("MathService", msg);
      }
   }
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter04/Windows_Service/MathLibrary/SimpleMath.vb
Public Class SimpleMath
   Inherits MarshalByRefObject

   Public Sub New()
      WriteLogEntry("SimpleMath ctor called")
   End Sub

   Public Function Add(n1 As Integer, n2 As Integer) As Integer
      WriteLogEntry(String.Format("SimpleMath.Add({0}, {1})", n1, n2))
      Return n1 + n2
   End Function

   Public Function Subtract(n1 As Integer, n2 As Integer) As Integer
      WriteLogEntry(String.Format("SimpleMath.Subtract({0}, {1})", n1, n2))
      Return n1 - n2
   End Function

   Public Shared Sub WriteLogEntry(msg As String)
      EventLog.WriteEntry("MathService", msg)
   End Sub

End Class